magpie/api/webhooks.py

```diff
@@ -4,65 +4,87 @@
 
 import requests
 import transaction
-from pyramid.threadlocal import get_current_registry
 from six.moves.urllib.parse import urlparse
 
 from magpie import models
 from magpie.api.schemas import UserStatuses
 from magpie.constants import get_constant
 from magpie.db import get_db_session_from_config_ini
 from magpie.register import get_all_configs
-from magpie.utils import ExtendedEnum, get_logger, get_settings, raise_log
+from magpie.utils import CONTENT_TYPE_JSON, FORMAT_TYPE_MAPPING, ExtendedEnum, get_logger, get_settings, raise_log
 
 if TYPE_CHECKING:
-    from typing import Dict, List, Optional
-
-    from magpie.typedefs import JSON, SettingsType, Str
+    from typing import List, Optional
+
+    from magpie.typedefs import (
+        AnySettingsContainer,
+        SettingsType,
+        Str,
+        WebhookConfig,
+        WebhookConfigSettings,
+        WebhookPayload,
+        WebhookTemplateParameters
+    )
 
 # List of keys that should be found for a single webhook item in the config
-WEBHOOK_KEYS = {
+WEBHOOK_KEYS_REQUIRED = {
     "name",
     "action",
     "method",
     "url",
     "payload"
 }
+WEBHOOK_KEYS_OPTIONAL = {
+    "format"
+}
+WEBHOOK_KEYS = WEBHOOK_KEYS_REQUIRED | WEBHOOK_KEYS_OPTIONAL
 
-# These are the parameters permitted to use the template form in the webhook payload.
-WEBHOOK_TEMPLATE_PARAMS = ["user_name", "tmp_url"]
+# These are *potential* parameters permitted to use the template form in the webhook payload.
+# Each parameter transferred to any given webhook are provided distinctively for each case.
+WEBHOOK_TEMPLATE_PARAMS = ["user_name", "user_id", "user_email", "callback_url"]
```

Comment:
Does user_id has a value for external use? Can it be used in the api? Since the primary ussage is in the database, if it is not exposed/expected in the API I would not provide it here neither.

Reply:
I was thinking maybe it could (should?) be used by cowbird to generate for example the private working directories. 
To do things properly, definitions of user-specific items external to Magpie should be anonymized.